File: src/main/java/com/matamercer/microblog/storage/StorageService.java
package com.matamercer.microblog.storage;

import org.springframework.core.io.Resource;
import org.springframework.web.multipart.MultipartFile;

import java.nio.file.Path;
import java.util.stream.Stream;

public interface StorageService {
    void init();

    void store(Path fileDestPath, MultipartFile file);

    Resource loadAsResource(Path filePath);



    void deleteAll();
}

File: src/main/java/com/matamercer/microblog/storage/StorageService.kt
package com.matamercer.microblog.storage

import org.springframework.core.io.Resource
import org.springframework.web.multipart.MultipartFile
import java.nio.file.Path

interface StorageService {
    fun init()
    fun store(fileDestPath: Path, file: MultipartFile)
    fun loadAsResource(filePath: Path): Resource
    fun delete(filePath: Path)
    fun deleteAll()
}